src/python_testing/TC_OPCREDS_3_9.py

```diff
@@ -0,0 +1,354 @@
+#
+#    Copyright (c) 2025 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+# See https://github.com/project-chip/connectedhomeip/blob/master/docs/testing/python.md#defining-the-ci-test-arguments
+# for details about the block below.
+#
+# === BEGIN CI TEST ARGUMENTS ===
+# test-runner-runs:
+#   run1:
+#     app: ${ALL_CLUSTERS_APP}
+#     app-args: --discriminator 1234 --KVS kvs1 --trace-to json:${TRACE_APP}.json
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --PICS src/app/tests/suites/certification/ci-pics-values
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+
+import enum
+import hashlib
+import logging
+from binascii import unhexlify
+from typing import Optional
+
+import chip.clusters as Clusters
+from chip.interaction_model import InteractionModelError, Status
+from chip.testing.matter_testing import MatterBaseTest, TestStep, async_test_body, default_matter_test_main
+from chip.tlv import TLVReader
+from chip.utils import CommissioningBuildingBlocks
+from ecdsa import NIST256p, VerifyingKey
+from ecdsa.keys import BadSignatureError
+from mobly import asserts
+
+
+class MatterCertParser:
+    SUBJECT_TAG = 6
+    SUBJECT_PUBLIC_KEY_TAG = 9
+    SUBJECT_FABRIC_ID_TAG = 21
+
+    def __init__(self, matter_cert_bytes: bytes):
+        self.parsed_tlv = TLVReader(matter_cert_bytes).get()["Any"]
+
+    def get_subject_names(self) -> dict[int, object]:
+        return {tag: value for tag, value in self.parsed_tlv[self.SUBJECT_TAG]}
+
+    def get_public_key_bytes(self) -> bytes:
+        public_key_bytes = self.parsed_tlv[self.SUBJECT_PUBLIC_KEY_TAG]
+        return public_key_bytes
+
+
+# From Matter spec src/crypto_primitives/crypto_primitives.py
+class MappingsV1(enum.IntEnum):
```

Comment:
using an enum feels off ... this seems to be like a namespace use. Can we just have these as constants? This is not an enumeration ... so if we really want contants maybe just use class without enum.